src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -22,50 +30,230 @@ const ActorInit Obj_Kibako2_InitVars = {
+        f32 cos = Math_CosS(angle);
+        f32 tempRand;
+        s32 phi_s0;
+
+        tempRand = Rand_ZeroOne() * 30.0f;
+        pos.x = sin * tempRand;
+        pos.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        pos.z = cos * tempRand;
+        velocity.x = pos.x * 0.2f;
+        velocity.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        velocity.z = pos.z * 0.2f;
+        pos.x += thisPos->x;
+        pos.y += thisPos->y;
+        pos.z += thisPos->z;
+        tempRand = Rand_ZeroOne();
+        if (tempRand < 0.05f) {
+            phi_s0 = 0x60;
+        } else if (tempRand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -200, phi_s0, 28, 2, 0, (Rand_ZeroOne() * 30.0f) + 5.0f,
+                             0, 0, 70, KAKERA_COLOR_NONE, OBJECT_KIBAKO2, D_06001040);
+    }
+    func_800BBFB0(globalCtx, thisPos, 90.0f, 6, 100, 160, 1);
+}
+
+void ObjKibako2_SpawnCollectible(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 collectible;
+
+    collectible = func_800A8150(KIBAKO2_COLLECTIBLE_ID(&this->dyna.actor));
+    if (collectible >= 0) {
+        Item_DropCollectible(globalCtx, &this->dyna.actor.world.pos,
+                             collectible | KIBAKO2_COLLECTIBLE_FLAG(&this->dyna.actor) << 8);
+    }
+}
+
+void ObjKibako2_SpawnSkulltula(ObjKibako2* this, GlobalContext* globalCtx) {
+    s16 yRotation;
+    s32 actorSpawnParam;
+    Actor* skulltula;
 
-extern ColliderCylinderInit D_8098EE60;
-extern InitChainEntry D_8098EE8C[];
+    if (ObjKibako2_ContainsSkulltula(this, globalCtx)) {
+        actorSpawnParam = KIBAKO2_SKULLTULA_SPAWN_PARAM(&this->dyna.actor);
+        yRotation = ((u32)Rand_Next() >> 0x11) + this->dyna.actor.yawTowardsPlayer + 0xC000;
+        skulltula =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_SW, this->dyna.actor.world.pos.x,
+                        this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, yRotation, 0, actorSpawnParam);
+        if (skulltula != NULL) {
+            skulltula->parent = &this->dyna.actor;
+            skulltula->velocity.y = 13.0f;
+            skulltula->speedXZ = 0.0f;
+        }
+    }
+}
 
-extern UNK_TYPE D_06000960;
-extern UNK_TYPE D_06000B70;
-extern UNK_TYPE D_06001040;
+void ObjKibako2_SpawnContents(ObjKibako2* this, GlobalContext* globalCtx) {
+    if (KIBAKO2_CONTENTS(&this->dyna.actor) == CONTENTS_COLLECTIBLE) {
+        ObjKibako2_SpawnCollectible(this, globalCtx);
+    } else {
+        ObjKibako2_SpawnSkulltula(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E5C0.s")
+void ObjKibako2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+    s32 pad;
+    ObjKibako2Contents contents;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E62C.s")
+    contents = KIBAKO2_CONTENTS(&this->dyna.actor);
```

Comment:
Git won't let me give suggestions when there's a deleted line in the middle, you similar to comments above, you can combing contents declaration with this line. Make sure there's still a space after the variable declaration :)